simulation/halsim_gui/src/main/native/cpp/Assembly2D.cpp

```diff
@@ -0,0 +1,249 @@
+                                           windowPos, colorLookUpTable[bodyConfig.color], bodyConfig, "");
+
+        // If the line has children then draw them with the stating points being the end of the parent
+        if (bodyConfig.children.size() != 0) {
+            buildDrawList(XEnd, YEnd, drawList, angle,
+                          bodyConfig.children, windowPos);
+        }
+    }
+}
+
+static void DisplayAssembly2D() {
+    ImVec2 windowPos = ImGui::GetWindowPos();
+    ImDrawList *drawList = ImGui::GetWindowDrawList();
+    buildDrawList(ImGui::GetWindowWidth() / 2, ImGui::GetWindowHeight(), drawList, 0,
+                  bodyConfigList, windowPos);
+}
+
+BodyConfig readSubJson(const std::string &name, wpi::json const &body) {
+    BodyConfig c;
+    try {
+        c.name = name + body.at("name").get<std::string>() + "/";
+    } catch (const wpi::json::exception &e) {
+        wpi::errs() << "could not read body name: " << e.what() << '\n';
+    }
+    try {
+        c.type = body.at("type").get<std::string>();
+    } catch (const wpi::json::exception &e) {
+        wpi::errs() << "camera '" << c.name << "': could not type path: " << e.what() << '\n';
+    }
+    try {
+        c.startLocation = body.at("startLocation").get<int>();
+    } catch (const wpi::json::exception &e) {
+        wpi::errs() << "startLocation '" << c.name << "': could not find startLocation path: " << e.what()
+                    << '\n';
+    }
+    try {
+        c.length = body.at("length").get<int>();
+    } catch (const wpi::json::exception &e) {
+        wpi::errs() << "length '" << c.name << "': could not find length path: " << e.what() << '\n';
+    }
+    try {
+        c.color = body.at("color").get<std::string>();
+    } catch (const wpi::json::exception &e) {
+        wpi::errs() << "color '" << c.name << "': could not find color path: " << e.what() << '\n';
+    }
+    try {
+        c.angle = body.at("angle").get<int>();
+    } catch (const wpi::json::exception &e) {
+        wpi::errs() << "angle '" << c.name << "': could not find angle path: " << e.what() << '\n';
+    }
+    try {
+        c.lineWidth = body.at("lineWidth").get<int>();
+    } catch (const wpi::json::exception &e) {
+        wpi::errs() << "lineWidth '" << c.name << "': could not find lineWidth path: " << e.what() << '\n';
+    }
+    try {
+        for (wpi::json const &child : body.at("children")) {
+            c.children.push_back(readSubJson(c.name, child));
+            wpi::outs() << "Reading Child with name " << c.name << '\n';
+        }
+    } catch (const wpi::json::exception &e) {
+        wpi::errs() << "could not read body: " << e.what() << '\n';
+    }
+    return c;
+}
+
+static std::list<BodyConfig> readJson(std::string jFile) {
+    std::list<BodyConfig> cList;
+    std::error_code ec;
+    std::string name;
+    wpi::raw_fd_istream is(jFile, ec);
+    if (ec) {
+        wpi::errs() << "could not open '" << jFile << "': " << ec.message() << '\n';
+    }
```

Comment:
If an error occurs in this function, it should immediately return an empty list rather than keep processing?